src_features/signMessageEIP712_common/common_712.c

```diff
@@ -8,62 +9,52 @@
+    CX_ASSERT(cx_hash_no_throw((cx_hash_t *) &global_sha3,
+                               0,
+                               (uint8_t *) EIP_712_MAGIC,
+                               sizeof(EIP_712_MAGIC),
+                               NULL,
+                               0));
+    CX_ASSERT(cx_hash_no_throw((cx_hash_t *) &global_sha3,
+                               0,
+                               tmpCtx.messageSigningContext712.domainHash,
+                               sizeof(tmpCtx.messageSigningContext712.domainHash),
+                               NULL,
+                               0));
+    CX_ASSERT(cx_hash_no_throw((cx_hash_t *) &global_sha3,
+                               CX_LAST,
+                               tmpCtx.messageSigningContext712.messageHash,
+                               sizeof(tmpCtx.messageSigningContext712.messageHash),
+                               hash,
+                               sizeof(hash)));
     PRINTF("EIP712 Domain hash 0x%.*h\n", 32, tmpCtx.messageSigningContext712.domainHash);
     PRINTF("EIP712 Message hash 0x%.*h\n", 32, tmpCtx.messageSigningContext712.messageHash);
-    io_seproxyhal_io_heartbeat();
-    os_perso_derive_node_bip32(CX_CURVE_256K1,
-                               tmpCtx.messageSigningContext712.bip32.path,
-                               tmpCtx.messageSigningContext712.bip32.length,
-                               privateKeyData,
-                               NULL);
-    io_seproxyhal_io_heartbeat();
-    cx_ecfp_init_private_key(CX_CURVE_256K1, privateKeyData, 32, &privateKey);
-    explicit_bzero(privateKeyData, sizeof(privateKeyData));
+
     unsigned int info = 0;
-    io_seproxyhal_io_heartbeat();
-    cx_ecdsa_sign(&privateKey,
-                  CX_RND_RFC6979 | CX_LAST,
-                  CX_SHA256,
-                  hash,
-                  sizeof(hash),
-                  signature,
-                  sizeof(signature),
-                  &info);
-    explicit_bzero(&privateKey, sizeof(privateKey));
+    if (bip32_derive_ecdsa_sign_rs_hash_256(CX_CURVE_256K1,
+                                            tmpCtx.messageSigningContext712.bip32.path,
+                                            tmpCtx.messageSigningContext712.bip32.length,
+                                            CX_RND_RFC6979 | CX_LAST,
+                                            CX_SHA256,
+                                            hash,
+                                            sizeof(hash),
+                                            G_io_apdu_buffer + 1,
+                                            G_io_apdu_buffer + 1 + 32,
+                                            &info) != CX_OK) {
+        THROW(0x6F00);
+    }
     G_io_apdu_buffer[0] = 27;
     if (info & CX_ECCINFO_PARITY_ODD) {
         G_io_apdu_buffer[0]++;
     }
     if (info & CX_ECCINFO_xGTn) {
         G_io_apdu_buffer[0] += 2;
     }
-    format_signature_out(signature);
```

Comment:
this did nothing ??

Reply:
cc @apaillier-ledger you might know more ?

Reply:
Yeap before we had:
- sign in signature with cx_ecdsa_sign
- extract r and s from signature and put it in G_io_apdu_buffer with format_signature_out

Now both are done in once with a single bip32_derive_ecdsa_sign_rs_hash_256 call.

Note that I didn't see the removal of the `format_signature_out`, can't it be removed now?

Reply:
`format_signature_out` has been removed, because no more called anywhere.